freeipa/src/test/java/com/sequenceiq/freeipa/controller/UserV1ControllerTest.java

```diff
@@ -115,14 +120,36 @@ void synchronizeAllUsers() {
         request.setMachineUsers(machineUsers);
 
         Operation operation = mock(Operation.class);
-        when(userSyncService.synchronizeUsersWithCustomPermissionCheck(any(), any(), any(), any(), any(), any())).thenReturn(operation);
+        when(userSyncService.synchronizeUsersWithCustomPermissionCheck(any(), any(), any(), any(), any())).thenReturn(operation);
         SyncOperationStatus status = mock(SyncOperationStatus.class);
         when(operationToSyncOperationStatus.convert(operation)).thenReturn(status);
 
         assertEquals(status, ThreadBasedUserCrnProvider.doAs(USER_CRN, () -> underTest.synchronizeAllUsers(request)));
 
+        UserSyncRequestFilter userSyncFilter = new UserSyncRequestFilter(users, machineUsers, Optional.empty());
         verify(userSyncService, times(1)).synchronizeUsersWithCustomPermissionCheck(ACCOUNT_ID, USER_CRN, environments,
-                users, machineUsers, AuthorizationResourceAction.DESCRIBE_ENVIRONMENT);
+                userSyncFilter, AuthorizationResourceAction.DESCRIBE_ENVIRONMENT);
+    }
+
+    @Test
+    void synchronizeAllUsersDeleteWorkloadUser() {
+        Set<String> users = Set.of(USER_CRN);
```

Comment:
I think we need a test that checks the parameter validation and ensures that we reject requests with a deleted workload username that contain 0 or >1 crns.